### ISSUE TITLE
[ubuntu] ofSystemLoadDialog doesn't work

### ISSUE THREAD

**Issue by jaysonh** [openframeworks/openFrameworks#7143]
I'm using Ubuntu 22.04 with OF version: of_v0.11.2_linux64gcc6_release

When I try running the command ofSystemLoadDialog nothing happens, no error messages or any other indications of a problem.

This is a simplified version illustrating the problem:

`#include "ofApp.h"

//--------------------------------------------------------------
void ofApp::setup(){

}

//--------------------------------------------------------------
void ofApp::update(){

}

//--------------------------------------------------------------
void ofApp::draw(){

}

//--------------------------------------------------------------
void ofApp::keyPressed(int key){
	if(key==' ')
	{

		ofFileDialogResult result = ofSystemLoadDialog("Load file");
		if(result.bSuccess) {
		  string path = result.getPath();
		  // load your file at `path`
		}
	}
}
`

**Comment by mchelifpv** [comment]
The same thing. With stable and nightly version OF. Ubuntu 22.04.1 LTS.

**Comment by ofTheo** [comment]
Thanks @jaysonh @mchelifpv 
These are the lines being called - it might  be worth checking that the lines are even executing? Maybe with some std::cout debug messages?

https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/utils/ofSystemUtils.cpp#L491-L496

**Comment by mchelifpv** [comment]
No, this block is not executed. Marked every line in this code:

       ofLogNotice() << "test 1";                                    
#if defined( TARGET_LINUX ) && defined (OF_USING_GTK)                 
                ofLogNotice() << "test 2";                            
                auto locale = std::locale();                          
                ofLogNotice() << "test 3";                            
                if(bFolderSelection) {                                
                        results.filePath = gtkFileDialog(GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER,windowTitle,ofToDataPath(defaultPath));
                        ofLogNotice() << "test 4";
                } else {
                        results.filePath = gtkFileDialog(GTK_FILE_CHOOSER_ACTION_OPEN,windowTitle,ofToDataPath(defaultPath));
                        ofLogNotice() << "test 5";
                }
                ofLogNotice() << "test 6";
                resetLocale(locale);
                ofLogNotice() << "test 7";
#endif
        ofLogNotice() << "test 8";

Here is the result:
/fileOpenSaveDialogExample
[notice ] test 1
[notice ] test 8
[verbose] User hit cancel: 
[notice ] test 1
[notice ] test 8
[verbose] User hit cancel:

**Comment by dimitre** [comment]
I don't have a linux machine here but maybe OF_USING_GTK is undefined
I've took a quick look in the file
config.linux.common.mk

and it seems to be defined here
```
PLATFORM_DEFINES =

# add OF_USING_GTK define IF we have it defined as a system library
ifeq ($(HAS_SYSTEM_GTK2),0)
	PLATFORM_DEFINES += OF_USING_GTK
endif
ifeq ($(HAS_SYSTEM_GTK3),0)
	PLATFORM_DEFINES += OF_USING_GTK
endif
```
which are defined by

```
	HAS_SYSTEM_GTK3 = $(shell pkg-config gtk+-3.0 --exists; echo $$?)
	HAS_SYSTEM_GTK2 = $(shell pkg-config gtk+-2.0 --exists; echo $$?)
```

so maybe it can be checked in terminal if any of this conditions is fulfilled

**Comment by ofTheo** [comment]
Ahh. Thanks @dimitre 
I think Ubuntu 22 has GTK4 which is why OF_USING_GTK is not being defined. 

Prob needs something like this:

```
	HAS_SYSTEM_GTK4 = $(shell pkg-config gtk+-4.0 --exists; echo $$?)
```

and 

```
ifeq ($(HAS_SYSTEM_GTK4),0)
	PLATFORM_DEFINES += OF_USING_GTK
endif
```

Or a more generic has any GTK check?

**Comment by ofTheo** [comment]
A quick temp fix @mchelifpv would be to do 

```
#define OF_USING_GTK 
```

in ofConstants.h

**Comment by dimitre** [comment]
Yeah, and I think if he can test your suggestion of adding the lines, maybe submitting a PR, so it is fixed on master :)

**Comment by mchelifpv** [comment]
Found a solution! Need to install the libgtk2.0-dev package. Please add it to the dependency checking script "scripts/linux/ubuntu/install_dependencies.sh".

**Comment by ofTheo** [comment]
Thanks! 

I think the better solution might be to detect gtk4
Can you see if you have it by doing: 

`pkg-config --list-all | grep gtk` 

and share the results?

**Comment by mchelifpv** [comment]
```
pkg-config --list-all | grep gtk
gtk+-2.0                       GTK+ - GTK+ Graphical UI Library (x11 target)
gtk+-unix-print-2.0            GTK+ - GTK+ Unix print support
gtk+-x11-2.0                   GTK+ - GTK+ Graphical UI Library (x11 target)
```

**Comment by ofTheo** [comment]
Thanks @mchelifpv - I think that should have been caught by these lines: 

https://github.com/openframeworks/openFrameworks/blob/master/scripts/linux/ubuntu/install_dependencies.sh#L140-L147

But I'll double check and we can add gtk4 support at the same time.

**Comment by dimitre** [comment]
I'm using multipass here to run Ubuntu 22.04 in a macOS machine
and running 
```
pkg-config --list-all | grep gtk
```
I get this result
```
gtk+-3.0                    GTK+ - GTK+ Graphical UI Library
gtk+-broadway-3.0           GTK+ - GTK+ Graphical UI Library
gtk+-unix-print-3.0         GTK+ - GTK+ Unix print support
gtk+-wayland-3.0            GTK+ - GTK+ Graphical UI Library
gtk+-x11-3.0                GTK+ - GTK+ Graphical UI Library
```
not sure if it was installed by OF or not
